### DISCUSSION TITLE
AES Decrypt not working in iOS 12 or lower

### DISCUSSION THREAD

**Discussion by DanielGZ95** [krzyzanowskim/CryptoSwift#949] [General]
**Describe the bug**
I'm trying to decypt data encrypted using AES-256. The decrypt process works fine on devices with iOS 13 and above. Only the devices with iOS 12 are returning invalid data when I call the decrypt method. The string that I'm trying to decrypt is generated inside a web service. 
The strange part is that when I encrypt the data inside of my project, the decrypt process works good. Is there a workaround or a solution that I can apply to my project? I need to support iOS 12. 

**Reproduce**
Steps to reproduce:

Xcode: 12.4
CryptoSwift: 1.3.2

```
let aes = try AES(key: keyString, iv: ivString, padding: .pkcs5)
let data = NSData(base64Encoded: encryptedString, options: NSData.Base64DecodingOptions(rawValue: 0))!
let dec = try aes.decrypt([UInt8](data)) //On iOS12: This returns corrupted data. iOS 13 works fine.
let decryptedData = Data(dec)
                
let final = String(bytes: decryptedData.bytes, encoding: .utf8) 
```

**Comment by krzyzanowskim** [comment]
Can you please check these for me:

1. if it happens in Release build on Simulator
2. if it happens in Release build on Device
3. if it happens in Debug build on Device

**Comment by krzyzanowskim** [comment]
on top of that, can you please test with CryptoSwift 1.3.8?

**Comment by DanielGZ95** [comment]
I already tested using CryptoSwift 1.3.8 and still not working. 
It happens in debug and release builds. I tested with simulator and using phisical device throught browserstack and in all devices the result is the same.

**Comment by krzyzanowskim** [comment]
Can you please share a sample Xcode project (with values where you see the issue) I could run on my end and see what's going on?

**Comment by DanielGZ95** [comment]
Sure, this is a sample app. All the methods that I'm using in my real project are in the AppDelegate. In this project I'm getting the same results. I hope you can help me with this please. And thanks!

[CryptoSwiftTest.zip](https://github.com/krzyzanowskim/CryptoSwift/files/6327156/CryptoSwiftTest.zip)

**Comment by DanielGZ95** [comment]
Hi @krzyzanowskim  did you got some time to check my test project?

**Comment by imrankst1221** [comment]
@DanielGZ95 did you find any solution regarding this? 
@krzyzanowskim fetching same issue version '1.4.1' not working on 12.4 (iPad pro 9.7-inch Simulator).

**Comment by DanielGZ95** [comment]
@imrankst1221 I stop using this library and I used the native way provided by swift. I based my implementation in this resource https://developer.apple.com/documentation/cryptokit/performing_common_cryptographic_operations
